luigi/scheduler.py

```diff
@@ -444,11 +444,12 @@ def fail_dead_worker_task(self, task, config, assistants):
 
     def update_status(self, task, config):
         # Mark tasks with no remaining active stakeholders for deletion
-        if not task.stakeholders:
-            if task.remove is None:
-                logger.debug("Task %r has stakeholders %r but none remain connected -> might remove "
-                             "task in %s seconds", task.id, task.stakeholders, config.remove_delay)
-                task.remove = time.time() + config.remove_delay
+        if (not task.stakeholders) and (task.remove is None) and (task.status != RUNNING):
+            # We don't check for the RUNNING case, because that is already handled
+            # by the fail_dead_worker_task function.
+            logger.debug("Task %r has no stakeholders anymore -> might remove "
```

Comment:
Sorry. The changed log message is not related to this patch. Just see it as a bonus update of legacy code.